src/js/serial_backend.js

```diff
@@ -697,75 +697,72 @@ function update_live_status() {
     });
 
     if (GUI.active_tab !== 'cli' && GUI.active_tab !== 'presets') {
-        MSP.send_message(MSPCodes.MSP_BOXNAMES, false, false);
-        if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_32)) {
-            MSP.send_message(MSPCodes.MSP_STATUS_EX, false, false);
-        } else {
-            MSP.send_message(MSPCodes.MSP_STATUS, false, false);
-        }
-        MSP.send_message(MSPCodes.MSP_ANALOG, false, false);
-    }
-
-    const active = ((Date.now() - FC.ANALOG.last_received_timestamp) < 300);
-
-    for (let i = 0; i < FC.AUX_CONFIG.length; i++) {
-        if (FC.AUX_CONFIG[i] === 'ARM') {
-            if (bit_check(FC.CONFIG.mode, i)) {
-                $(".armedicon").addClass('active');
-            } else {
-                $(".armedicon").removeClass('active');
-            }
-        }
-        if (FC.AUX_CONFIG[i] === 'FAILSAFE') {
-            if (bit_check(FC.CONFIG.mode, i)) {
-                $(".failsafeicon").addClass('active');
-            } else {
-                $(".failsafeicon").removeClass('active');
+        Promise.resolve(true)
+        .then(() => MSP.promise(MSPCodes.MSP_BOXNAMES))
+        .then(() => MSP.promise(semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_32) ? MSPCodes.MSP_STATUS_EX : MSPCodes.MSP_STATUS))
+        .then(() => MSP.promise(MSPCodes.MSP_ANALOG))
+        .then(() => {
+            const active = ((Date.now() - FC.ANALOG.last_received_timestamp) < 300);
+
+            for (let i = 0; i < FC.AUX_CONFIG.length; i++) {
+                if (FC.AUX_CONFIG[i] === 'ARM') {
+                    if (bit_check(FC.CONFIG.mode, i)) {
+                        $(".armedicon").addClass('active');
+                    } else {
+                        $(".armedicon").removeClass('active');
+                    }
+                }
+                if (FC.AUX_CONFIG[i] === 'FAILSAFE') {
+                    if (bit_check(FC.CONFIG.mode, i)) {
+                        $(".failsafeicon").addClass('active');
+                    } else {
+                        $(".failsafeicon").removeClass('active');
+                    }
+                }
```

Comment:
I keep seeing this pattern. But what is the value of this? Especially `Promise.resolve(true)`

I think you could achieve similar flow with using `async` on the function and then just `await`ing these calls instead.

Reply:
Fixed.